mods/persistence/controllers/subsystems/persistence.dm

```diff
@@ -1,679 +1,278 @@
-	var/in_loaded_world 	= 	FALSE	// Whether or not we're in a world that was loaded.
+	// *** Config ***
+	///Whether or not rent will be required for created sectors.
+	var/rent_enabled = FALSE
+	///If set, we'll do everything we can to generate a valid save even if some things do runtime.
+	var/error_tolerance = PERSISTENCE_ERROR_TOLERANCE_NONE
 
-	var/list/saved_areas	= 	list()
-	var/list/saved_levels 	= 	list()	// Saved levels are saved entirely and optimized with get_base_turf()
-	var/list/saved_extensions = list()  // Extensions mark themselves to be saved on world save.
-
-	var/rent_enabled = FALSE			// Whether or not rent will be required for created sectors.
-
-	var/serializer/sql/serializer = new() // The serializer impl for actually saving.
-	var/serializer/sql/one_off/one_off	= new() // The serializer impl for one off serialization/deserialization.
-
-	var/list/limbo_removals = list() // Objects which will be removed from limbo on the next save. Format is list(limbo_key, limbo_type)
-	var/list/limbo_refs = list()	 // Objects which are deserialized out of limbo don't have their refs in the database immediately, so we add them here until the next save
-									 // Format is p_id -> ref
+	// *** State ***
 
+	///Whether or not a save exists in the DB. Value is cached when a save did exist at runtime for performance reasons.
+	var/save_exists     = FALSE
+	///Whether or not we're currently running a loaded save.
+	var/in_loaded_world = FALSE
+	///Whether we're currently loading a world save.
 	var/loading_world = FALSE
+	///The time when the currently loaded world save was made if any. NULL means we never loaded a save.
+	var/loaded_save_time
+	///Save log entry index from the database for the currently running save. Used to fill in the saving result into the db.
+	var/save_log_id
+	///Holds the previous entering allowed state while we're running a save.
+	var/was_entering_allowed
+
+	///Text to log into the database log for the completion/failure of the save
+	var/save_complete_text = ""
+	///Span class to use for displaying the completion/failure message into chat.
+	var/save_complete_span_class = "danger"
+	///Amount of z-levels saved
+	var/nb_saved_z_levels = 0
+	///Amount of atoms saved
+	var/nb_saved_atoms    = 0
+
+	//#FIXME: Not sure if keeping a separate list of those here is a very good idea?
+	var/list/saved_areas      = list()
+	/// Saved levels are saved entirely and optimized with get_base_turf()
+	var/list/saved_levels     = list()
+	/// Extensions mark themselves to be saved on world save.
+	var/list/saved_extensions = list()
+
+	//#FIXME: The idea for having a generic serializer class was that we wouldn't initiate the exact subtype in the var definition.
+	//        So that you could just interchangeably use another kind of serializer that saves to json or xml or whatever instead of sql seamlessly.
+	//        This implementation completely nullifies the whole point of having a serializer class currently?
```

Comment:
Theoretically this is true, but I doubt we'll ever reach the point where the implementation is completely generic. Due to proc call overhead, there's also the chance that over generifying things will hit performance. There is definitely repeated code in the subtypes though that should be fixed.

Reply:
Yeah, we're not going to have a need for something like that. But it's just food for thoughts.